applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_plane_strain_2d.cpp

```diff
@@ -404,37 +308,26 @@ void LinearJ2PlasticityPlaneStrain2D::GetLawFeatures(Features& rFeatures)
     rFeatures.mSpaceDimension = 2;
 }
 
-int LinearJ2PlasticityPlaneStrain2D::Check(const Properties& rMaterialProperties,
-                                           const GeometryType& rElementGeometry,
-                                           const ProcessInfo& rCurrentProcessInfo)
-{
-    KRATOS_CHECK(rMaterialProperties.Has(YOUNG_MODULUS));
-    KRATOS_CHECK(rMaterialProperties.Has(POISSON_RATIO));
-    KRATOS_CHECK(rMaterialProperties.Has(YIELD_STRESS));
-    KRATOS_CHECK(rMaterialProperties.Has(REFERENCE_HARDENING_MODULUS));
-    KRATOS_CHECK(rMaterialProperties.Has(ISOTROPIC_HARDENING_MODULUS));
-    KRATOS_CHECK(rMaterialProperties.Has(INFINITY_HARDENING_MODULUS));
-    KRATOS_CHECK(rMaterialProperties.Has(HARDENING_EXPONENT));
-
-    return 0;
-}
+//************************************************************************************
+//************************************************************************************
 
 void LinearJ2PlasticityPlaneStrain2D::save(Serializer& rSerializer) const
 {
-    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, ConstitutiveLaw);
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, LinearJ2Plasticity3D);
     rSerializer.save("mInelasticFlag", mInelasticFlag);
-    rSerializer.save("mStrainEnergy", mStrainEnergy);
     rSerializer.save("mPlasticStrain", mPlasticStrain);
     rSerializer.save("mPlasticStrainOld", mPlasticStrainOld);
     rSerializer.save("mAccumulatedPlasticStrain", mAccumulatedPlasticStrain);
     rSerializer.save("mAccumulatedPlasticStrainOld", mAccumulatedPlasticStrainOld);
 }
 
+//************************************************************************************
+//************************************************************************************
+
 void LinearJ2PlasticityPlaneStrain2D::load(Serializer& rSerializer)
 {
-    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, ConstitutiveLaw);
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, LinearJ2Plasticity3D);
     rSerializer.load("mInelasticFlag", mInelasticFlag);
```

Comment:
From what I can see all of these variables are already serialized in the Baseclass (i.e. the 3d law)
If you don't add new members in this class then the serializer function is empty except the call to serialize the base class

Reply:
you are right. done.